crates/ide_completion/src/completions/flyimport.rs

```diff
@@ -1077,4 +1079,52 @@ fn main() {
             "#]],
         );
     }
+
+    #[test]
+    fn no_flyimports_in_traits_and_impl_declarations() {
```

Comment:
Maybe we could test this in `completions.rs` for all completion types? I.e. instead of potentially adding a test to each completion type that says "this shouldn't appear in impl declarations", have one test that calls all completion provider and checks that only the expected ones give results

Reply:
Sounds legit. 
I was concentrated on the tiny flyimport area fix and it felt odd to place the test somewhere else.

I'll put a comment into the original issue.